'''
Data is already in a good format(.jpg) so it does not need to be cleaned.  Also, given it isall a complete image no missing data exists.
'''

In [10]:
from PIL import Image
import numpy as np
import os
import random as rd


In [16]:
paths = ['HAM10000_images_part_1', 'HAM10000_images_part_2',
         'ISIC2018_Task3_Test_Images'] # 3 folders holding every image

png_list = []

for path in paths :
    # add folder image is in, '/', and png name to get pathway to all images
    png_list += ['./' + path + '/' + x for x in os.listdir(path)]

# test if all 11527 images are accounted for
print(len(png_list))





11527


In [25]:
# 1 random image from each folder
images = [png_list[rd.randint(0, 4999)], png_list[rd.randint(5000, 10013)], png_list[rd.randint(10014, 11526)]]


for image in images :
    # image of skin
    skin_img = Image.open(image)
    # array of skin image
    array_img = np.array(skin_img)
    
    print(image)
    skin_img.show()
    print(array_img)
    print('image shape : ', skin_img.size, '   image type : ', skin_img.mode, '  array shape : ', array_img.shape)
    print()




./HAM10000_images_part_1/ISIC_0028638.jpg
[[[226 169 175]
  [230 169 177]
  [229 168 176]
  ...
  [225 162 179]
  [220 162 177]
  [212 160 173]]

 [[228 167 174]
  [227 166 174]
  [226 169 176]
  ...
  [226 161 181]
  [226 165 181]
  [222 161 176]]

 [[228 169 173]
  [224 168 171]
  [228 169 175]
  ...
  [228 167 182]
  [226 168 180]
  [224 168 181]]

 ...

 [[203 165 156]
  [203 165 156]
  [201 162 155]
  ...
  [226 180 182]
  [225 179 181]
  [224 179 186]]

 [[201 161 153]
  [204 163 159]
  [206 165 159]
  ...
  [221 179 181]
  [221 181 181]
  [222 180 182]]

 [[207 162 156]
  [210 162 158]
  [208 163 157]
  ...
  [217 175 176]
  [225 181 182]
  [223 179 178]]]
image shape :  (600, 450)    image type :  RGB   array shape :  (450, 600, 3)

./HAM10000_images_part_2/ISIC_0033919.jpg
[[[167 163 177]
  [171 163 178]
  [173 162 179]
  ...
  [171 168 179]
  [166 169 178]
  [168 165 174]]

 [[167 163 177]
  [167 163 177]
  [172 161 175]
  ...
  [174 166 177]
  [173 168 175]
  [172 164 177]]



(eog:646370): EOG-WARNING **: 11:48:29.407: Generating thumbnail failed: Child process exited with code 1

(eog:646370): EOG-WARNING **: 11:48:29.461: Thumbnail creation failed


All 11527 images have file locations pointed to in png_list.  To show the files could be turned into numpy arrays for use in machine learning 3 png files each randomly chosen from 1 of 3 different directories with png files were used.  Their array values were shown with the (450, 600, 3) array size matching the 600 by 450 image size.  Also, the 3 in the array represented the 3 values in RBG format indicating the color of a pixel.  